src/relic.cpp

```diff
@@ -434,7 +434,13 @@ bool process_recharge_entry( item &itm, const relic_recharge &rech, Character &c
             std::abort();
         }
     }
-    itm.charges = clamp( itm.charges + rech.rate, 0, itm.ammo_capacity() );
+    // If relic has a valid ammo type, make sure the first charge loaded isn't a "none"
+    if( !itm.ammo_types().empty() && itm.charges == 0 ) {
+        itm.charges = clamp( itm.charges + rech.rate, 0, itm.ammo_capacity() );
+        itm.ammo_set( itm.ammo_default(), itm.charges );
+    } else {
+        itm.charges = clamp( itm.charges + rech.rate, 0, itm.ammo_capacity() );
+    }
```

Comment:
The line
> itm.charges = clamp( itm.charges + rech.rate, 0, itm.ammo_capacity() );

should be moved outside of the `if` clause, before or after it.
The line is repeated in both branches and is executed regardless of condition in `if (...)`, meaning it's just code repetition for no gain.

Reply:
Main reason I did it this way is, far as I can tell from reading the function, `ammo_set` won't set ammo type unless it has charges, so it has to be called after a charge has been loaded. In addition, because it sets the ammo to the default, it also has to be set to only call in a situation where there's no chance it will override non-default ammo with the default (which means it can only be called when we can be certain the item's currently empty, which can only be before the recharge action).

Reply:
Well, then it has to be moved before the if, like so:
```suggestion
    // If relic has a valid ammo type, make sure the first charge loaded isn't a "none"
    itm.charges = clamp( itm.charges + rech.rate, 0, itm.ammo_capacity() );
    if( !itm.ammo_types().empty() && itm.charges == 0 ) {
        itm.ammo_set( itm.ammo_default(), itm.charges );
    }
```
There is zero functional difference between original code and this suggestion, but the duplication is no more.

Reply:
Actually no, what if we tried this instead:
```suggestion
    itm.ammo_set( itm.ammo_default(), itm.charges + rech.rate );
```
Looking at `ammo_set`, it has internal check for null ammo_default and also updates `itm.charges`, so it may not be needed to update it directly.

Reply:
The main thing is, what does `ammo_set` do if you:
1. Load it with ammo that's not the default
2. Then call it with `ammo_default` as the first arg?

The issue is basically, if it's not set to double-check like the if statement does, then if you load sinew for example calling ammo_set will just override it with thread, or other examples of it transmutating pre-loaded ammo that isn't the default and turning it into the default.

It's not strictly nessecary for what I'm doing in Arcana since the only use of it won't have more than one ammo item per ammotype, but this would be needed for Magiclysm for stuff like the recharging sewing kit.

Reply:
Okay yeah, compile-tested the latter method proposed, loaded sinew into the modified test relic, let it charge a bit, and unloaded thread from it.